save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,267 @@
+/**
+ * A view for registration
+ */
+
+@file:Suppress("FILE_WILDCARD_IMPORTS", "WildcardImport")
+
+package com.saveourtool.save.frontend.components.views
+
+import com.saveourtool.save.domain.ImageInfo
+import com.saveourtool.save.frontend.components.basic.InputTypes
+import com.saveourtool.save.frontend.components.basic.inputTextFormRequired
+import com.saveourtool.save.frontend.http.getUser
+import com.saveourtool.save.frontend.utils.*
+import com.saveourtool.save.frontend.utils.classLoadingHandler
+import com.saveourtool.save.info.UserInfo
+import com.saveourtool.save.utils.AvatarType
+import com.saveourtool.save.v1
+import com.saveourtool.save.validation.FrontendRoutes
+import com.saveourtool.save.validation.isValidName
+
+import csstype.ClassName
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.asList
+import org.w3c.fetch.Headers
+import org.w3c.xhr.FormData
+import react.*
+import react.dom.aria.ariaLabel
+import react.dom.events.ChangeEvent
+import react.dom.html.ButtonType
+import react.dom.html.InputType
+import react.dom.html.ReactHTML
+
+import kotlinx.browser.window
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * `Props` retrieved from router
+ */
+external interface RegistrationProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userName: String?
+}
+
+/**
+ * [State] of registration view component
+ */
+external interface RegistrationViewState : State {
+    /**
+     * Conflict error message
+     */
+    var conflictErrorMessage: String?
+
+    /**
+     * Flag to handle uploading a file
+     */
+    var isUploading: Boolean
+
+    /**
+     * Image to owner avatar
+     */
+    var image: ImageInfo?
+
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+
+    /**
+     * Validation of input fields
+     */
+    var isValidUserName: Boolean?
+}
+
+/**
+ * A Component for registration view
+ */
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+class RegistrationView : AbstractView<RegistrationProps, RegistrationViewState>() {
+    private val fieldsMap: MutableMap<InputTypes, String> = mutableMapOf()
```

Comment:
I strongly recommend you to use state here instead of private field of class as far as it is more react style (afaik)

Reply:
Fixed